src/Compilers/VisualBasic/Test/Semantic/Semantics/OverloadResolutionPriorityTests.vb

```diff
@@ -0,0 +1,987 @@
+    If(i1First, i2Source, i1Source) + "
+End Class
+"
+
+            Dim source = "
+public class Program 
+    Shared Sub Main
+        Dim i3 As I3 = Nothing
+        C.M(i3) = 0
+    End Sub
+End Class
+"
+
+            Dim comp1 = CreateCompilation({source, reference, OverloadResolutionPriorityAttributeDefinitionVB}, options:=TestOptions.DebugExe)
+
+            Dim validate = Sub([module] As ModuleSymbol)
+                               Dim c = [module].ContainingAssembly.GetTypeByMetadataName("C")
+                               Dim ms = c.GetMembers("M").Cast(Of PropertySymbol)()
+                               For Each m In ms
+                                   Assert.Equal(If(m.Parameters(0).Type.Name = "I1", 1, 0), m.OverloadResolutionPriority)
+                               Next
+                           End Sub
+
+            CompileAndVerify(comp1, expectedOutput:="1", sourceSymbolValidator:=validate, symbolValidator:=validate).VerifyDiagnostics()
+
+            Dim comp2 = CreateCompilation(source, references:={comp1.ToMetadataReference()}, options:=TestOptions.DebugExe)
+            CompileAndVerify(comp2, expectedOutput:="1").VerifyDiagnostics()
+
+            Dim comp3 = CreateCompilation(source, references:={comp1.EmitToImageReference()}, options:=TestOptions.DebugExe)
+            CompileAndVerify(comp3, expectedOutput:="1").VerifyDiagnostics()
+        End Sub
+
+        <Fact>
+        Public Sub ParameterlessProperty_01()
+            Dim compilationDef = "
+Module Module1
+
+    Sub Main()
+        M1 = 0
+    End Sub
+
+    <System.Runtime.CompilerServices.OverloadResolutionPriority(-1)>
+    WriteOnly Property M1 As Integer
+        Set
+            System.Console.Write(1)
+        End Set
+    End Property
+
+    WriteOnly Property M1(Optional x As Integer = 0) As Integer
+        Set
+            System.Console.Write(2)
+        End Set
+    End Property
+End Module
+"
+            Dim compilation = CreateCompilation({compilationDef, OverloadResolutionPriorityAttributeDefinitionVB}, options:=TestOptions.ReleaseExe)
+            CompileAndVerify(compilation, expectedOutput:="2")
+        End Sub
+
+        <Fact>
+        Public Sub ParameterlessProperty_02()
+            Dim compilationDef = "
+Module Module1
+
+    Sub Main()
+        M1 = 0
+    End Sub
+
+    WriteOnly Property M1 As Integer
+        Set
+            System.Console.Write(1)
+        End Set
+    End Property
+
+    <System.Runtime.CompilerServices.OverloadResolutionPriority(1)>
+    WriteOnly Property M1(Optional x As Integer = 0) As Integer
+        Set
+            System.Console.Write(2)
+        End Set
+    End Property
+End Module
+"
+            Dim compilation = CreateCompilation({compilationDef, OverloadResolutionPriorityAttributeDefinitionVB}, options:=TestOptions.ReleaseExe)
+            CompileAndVerify(compilation, expectedOutput:="2")
+        End Sub
+
+        <Fact>
+        Public Sub DefaultProperty_01()
```

Comment:
>DefaultProperty_01 [](http://example.com/codeflow?start=19&length=18)

Is this test affected by the compiler change? Should `Main()` contain a call to the default property?

Reply:
> Is this test affected by the compiler change? 

No. But the test demonstrates that there are no interesting scenarios when parameter-less property is marked default

> Should ```Main()``` contain a call to the default property?

I do not find it interesting, given the declaration error